features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/NewOverview/NewCustomizedStepper.jsx

```diff
@@ -0,0 +1,427 @@
+import Typography from '@material-ui/core/Typography';
+
+
+const ColorlibConnector = withStyles({
+    alternativeLabel: {
+        top: 22,
+    },
+    active: {
+        '& $line': {
+            backgroundImage:
+                'linear-gradient(to left, #50BCEC 50%, #B1D31E 50%)',
+        },
+    },
+    completed: {
+        '& $line': {
+            backgroundImage:
+                'linear-gradient( #B1D31E, #B1D31E)',
+        },
+    },
+    line: {
+        height: 3,
+        border: 0,
+        backgroundColor: '#eaeaf0',
+        borderRadius: 1,
+    },
+})(StepConnector);
+
+const useColorlibStepIconStyles = makeStyles({
+    root: {
+        backgroundColor: '#ccc',
+        zIndex: 1,
+        color: '#fff',
+        width: 56,
+        height: 56,
+        display: 'flex',
+        borderRadius: '50%',
+        justifyContent: 'center',
+        alignItems: 'center',
+        border: '6px solid #E2E2E2',
+    },
+    active: {
+        backgroundColor: '#50BCEC',
+        border: '6px solid #E2E2E2',
+    },
+    completed: {
+        backgroundColor: '#B1D31E',
+        border: '6px solid #E2E2E2',
+    },
+});
+
+function ColorlibStepIcon(props) {
+    const classes = useColorlibStepIconStyles();
+    const { active, completed } = props;
+
+    return (
+        <div
+            className={clsx(classes.root, {
+                [classes.active]: active,
+                [classes.completed]: completed,
+            })}
+        />
+    );
+}
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        width: '100%',
+    },
+    button: {
+        marginRight: theme.spacing(1),
+    },
+    instructions: {
+        marginTop: theme.spacing(1),
+        marginBottom: theme.spacing(1),
+    },
+    iconTrue: {
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        backgroundColor: '#B1D31E',
+        zIndex: 1,
+        color: '#fff',
+        width: 15,
+        height: 15,
+        borderRadius: '50%',
+    },
+    iconFalse: {
+        color: '#fff',
+        display: 'block',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+        backgroundColor: grey[500],
+        zIndex: 1,
+        width: 15,
+        height: 15,
+        borderRadius: '50%',
+    },
+}));
+
+function getSteps() {
+    return ['Develop', 'Deploy', 'Test', 'Publish'];
+}
+
+export default function NewCustomizedStepper() {
+    const classes = useStyles();
+    const [api, updateAPI] = useAPI();
+    const [isUpdating, setUpdating] = useState(false);
+    const [deploymentsAvailable, setDeploymentsAvailable] = useState(false);
+    const [lifecycleState, setlifecycleState] = useState(null);
+    const isEndpointAvailable = api.endpointConfig !== null;
+    const isTierAvailable = api.policies.length !== 0;
+    const steps = getSteps();
+
+    let activeStep = 0;
+    if (api && (api.type === 'WEBSUB' || isEndpointAvailable) && isTierAvailable && !deploymentsAvailable) {
```

Comment:
Shall we move this api types to api.js , ideally there should be a method such as `isWebSub` to check this type